### PR TITLE
Demo usage of C# with DotNet8

### PR DESCRIPTION
**Pull request by andreaTP**
C# it's a single Wasi function away :slightly_smiling_face: 

(the size of the resulting binary is huge ...)

### CODE REVIEW

File: wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java [hunk 1]
```diff
@@ -170,9 +170,26 @@ public HostFunction clockResGet() {
...
-                    throw new WASMRuntimeException(
-                            "We don't yet support this WASI call: clock_res_get");
-                    // return new Value[] { Value.i32(0) };
+                    int clockId = args[0].asInt();
+                    int resultPtr = args[1].asInt();
+
+                    Memory memory = instance.memory();
+                    switch (clockId) {
+                        case WasiClockId.REALTIME:
+                            memory.writeLong(resultPtr, 1L);
```
Comment (andreaTP) [inline wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java:179]: I have done some reading, but I cannot come up with anything better than "just use 1" ...
I'm open to suggestions!

**Comment by andreaTP** [comment]
> (the size of the resulting binary is huge ...)

added a bunch of trimming options and now looks better.

File: wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java [hunk 2]
```diff
@@ -170,9 +170,26 @@ public HostFunction clockResGet() {
...
+                            return wasiResult(WasiErrno.ESUCCESS);
+                        case WasiClockId.MONOTONIC:
```
Comment (electrum) [inline wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java:181]: These two cases can be combined
```java
case WasiClockId.REALTIME:
case WasiClockId.MONOTONIC:

File: wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java (hunk 1 above)
Comment (electrum) [reply wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java:179]: This seems fine to me.

**Review by electrum — APPROVED** [review]
Cool!

File: wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java (hunk 2 above)
Comment (andreaTP) [reply wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java:179]: fixed

**Pull request merged**